lib/astutils.cpp

```diff
@@ -365,9 +365,9 @@ bool isOppositeExpression(bool cpp, const Token * const tok1, const Token * cons
         return false;
     if (isOppositeCond(true, cpp, tok1, tok2, library, pure))
         return true;
-    if (tok1->str() == "-" && !tok1->astOperand2())
+    if (tok1->str() == "-" && !(tok1->astOperand2() && tok1->astOperand1()))
```

Comment:
I do no understand this fix. Do you think that the AST is correct here?

For a "-" there should always be a `astOperand1()` as far as I know.

If the AST is wrong, please let me know I can try to fix it.

Reply:
I think you are right. The real fix is to change `tok1` to `tok2` for the second comparison. I can check just `astOperand1`.